### PR TITLE
Feature/59-Extend-documentation

### PR DESCRIPTION
**Pull request by NoB0**
- Add installation, usage pages following README
- Propose update for agent set up page

### CODE REVIEW

File: docs/source/configuration.rst [hunk 1]
```diff
@@ -1,36 +1,55 @@
...
+  output_name: "moviebot"
+  # By default, the agent has an HTTP API.
+  agent_uri: "http://127.0.0.1:5001"
+  domain: data/domains/movies.yaml
+  intents: data/interaction_models/crs_v1.yaml
+  items: data/movielens-25m-sample/movies_w_keywords.csv
+  ratings: data/movielens-25m-sample/ratings.csv
+  dialogues: data/agents/moviebot/annotated_dialogues.json
+  intent_classifier: "cosine"
+  # If using the diet classifier the following file needs to be provided.
```
Comment (kbalog) [inline docs/source/configuration.rst:54]: Nit: DIET allcaps

File: docs/source/usage.rst [hunk 2]
```diff
@@ -0,0 +1,38 @@
+Usage
+=====
+
+A YAML configuration file is necessary to start the MovieBot; see `default configuration <https://github.com/iai-group/UserSimCRS/blob/main/config/default/config_default.yaml>`_ for an example.
```
Comment (kbalog) [inline docs/source/usage.rst:4]: Nit: "MovieBot" => "IAI MovieBot"

File: docs/source/usage.rst [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  * Download the IAI MovieBot `here <https://github.com/iai-group/MovieBot/>`_
+  * Checkout to the 'separate-flask-server' branch
+  * Follow the IAI MovieBot installation instructions
+  * Start the IAI MovieBot locally: 
+  
+.. code-block:: shell
+    
+    python -m run_bot -c config/moviebot_config_no_integration.yaml`
+
+Note: you need to update the parameter `agent_uri` in the configuration in case MovieBot does not run on the default URI (i.e., `http://127.0.0.1:5001`).
```
Comment (kbalog) [inline docs/source/usage.rst:28]: "you need to update the parameter xxx" => "parameter xxx needs to be updated"

File: docs/source/usage.rst [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+Example
+-------
+
+This example shows how to run simulation using the default configuration and the `IAI MovieBot <https://github.com/iai-group/MovieBot>`_ as the conversational agent.
+
+1. Start IAI MovieBot locally
+
+  * Download the IAI MovieBot `here <https://github.com/iai-group/MovieBot/>`_
+  * Checkout to the 'separate-flask-server' branch
+  * Follow the IAI MovieBot installation instructions
```
Comment (kbalog) [inline docs/source/usage.rst:21]: Nit: closing periods.

**Review by kbalog — APPROVED** [review]
LGTM

There is a number of small comments, but no additional review round is needed.

File: docs/config.md [hunk 5]
```diff
@@ -1,25 +1,34 @@
...
+* agent_id: id of the agent tested.
+* output_name: specifies the output name for the simulation configuration that will be stored under `data/runs` at the end of the simulation
+* agent_uri: URI to communicate with the agent. By default we assume that the agent has an HTTP API
+* domain: A YAML file with domain-specific slot names
+* intents: Path to the intent schema file
+* items: Path to items file
+* ratings: Path to ratings file
+* dialogues: Path to domain config file
+* intent_classifier: Intent classifier model to be used. Only supports DialogueKit intent classifier
+* rasa_dialogues: File with Rasa annotated dialogues. Only needed when using a DIET intent classifier
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/config.md]: Done.

File: docs/config.md [hunk 6]
```diff
@@ -1,25 +1,34 @@
...
+output_name: "moviebot"
+# By default, the agent has an HTTP API.
+agent_uri: "http://127.0.0.1:5001"
+domain: data/domains/movies.yaml
+intents: data/interaction_models/crs_v1.yaml
+items: data/movielens-25m-sample/movies_w_keywords.csv
+ratings: data/movielens-25m-sample/ratings.csv
+dialogues: data/agents/moviebot/annotated_dialogues.json
+intent_classifier: "cosine"
+# If using the diet classifier the following file needs to be provided.
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/config.md]: Done.

File: docs/source/configuration.rst [hunk 7]
```diff
@@ -1,36 +1,55 @@
-Configuration
-=============
+Setting up a user simulator
+----------------------------
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/configuration.rst]: Done.

File: docs/source/configuration.rst [hunk 8]
```diff
@@ -1,36 +1,55 @@
...
 
-The agenda based simulator has a number of parameters that can be customized.
-These can be either provided in a config file and/or via the command line. Note that arguments passed through the command line will override those in the config file.
+Requirements
+============
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/configuration.rst]: Done.

File: docs/source/configuration.rst [hunk 9]
```diff
@@ -1,36 +1,55 @@
...
 
-Example
--------
+To run the simulation, you will need the following:
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/configuration.rst]: Done.

File: docs/source/configuration.rst [hunk 10]
```diff
@@ -1,36 +1,55 @@
...
 
-The configuration parameters are expected to be defined in a YAML file.
-TODO: Add args for context and persona.
+1. A YAML file with domain-specific slot names that will be used for the preference model.
+2. A CSV file containing the item collection. This file must at least contain 2 columns labeled *ID* and *NAME*.
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/configuration.rst]: Done.

File: docs/source/configuration.rst [hunk 11]
```diff
@@ -1,36 +1,55 @@
...
+3. A CSV file containing the item ratings in the shape of user ID, item ID, and rating triples.
+4. A YAML file containing the users’ and agent’s intent space, as well as the set of expected agents for each user intent, is required for the interaction model. The CIR6 interaction model shipped with library offers a baseline starting point, which may be further tailored according to the behavior and capabilities of the target CRS.
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/configuration.rst]: Done.

File: docs/source/configuration.rst [hunk 12]
```diff
@@ -1,36 +1,55 @@
...
+5. A small sample of annotated dialogues with the target CRS. The sample size depends on the complexity of the system, in terms of action space and language variety, but is generally in the order of 5-50 dialogues. The sample of dialogues must contain utterance-level annotations in terms of intents and entities, as this is required to train the NLU and NLG components. Note that the slots used for annotation should be the same as the ones defined in the domain file (1) and intents should follow the ones defined in the interaction model (4)
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/configuration.rst]: Done.

File: docs/source/configuration.rst [hunk 13]
```diff
@@ -1,36 +1,55 @@
...
+Requirements
+============
 
-Example
--------
+To run the simulation, you will need the following:
 
-The configuration parameters are expected to be defined in a YAML file.
-TODO: Add args for context and persona.
+1. A YAML file with domain-specific slot names that will be used for the preference model.
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/configuration.rst]: Done.

File: docs/source/configuration.rst [hunk 14]
```diff
@@ -1,36 +1,55 @@
...
-  
-  ontology: ontology. yaml
-  items: movies.csv
-  ratings: ratings.csv
-  annotated_dialogues: annotated_dialogues.json
-  interaction_model: cir6.yaml
-  context: 20
-  persona: 10
+Configuration
+=============
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/configuration.rst]: Done.

File: docs/source/configuration.rst [hunk 15]
```diff
@@ -1,36 +1,55 @@
...
+* agent_id: id of the agent tested.
+* output_name: specifies the output name for the simulation configuration that will be stored under `data/runs` at the end of the simulation
+* agent_uri: URI to communicate with the agent. By default we assume that the agent has an HTTP API
+* domain: A YAML file with domain-specific slot names
+* intents: Path to the intent schema file
+* items: Path to items file
+* ratings: Path to ratings file
+* dialogues: Path to domain config file
+* intent_classifier: Intent classifier model to be used. Only supports DialogueKit intent classifier
+* rasa_dialogues: File with Rasa annotated dialogues. Only needed when using a DIET intent classifier
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/configuration.rst]: Done.

File: docs/source/configuration.rst [hunk 16]
```diff
@@ -1,36 +1,55 @@
...
 
-  python -m usersimcrs.run_simulation -config configs/agenda_based.yaml
+Configuration example
+^^^^^^^^^^^^^^^^^^^^^
 
-* Overriding parameters via the command line:
+Below you have the default configuration to run simulation with the IAI MovieBot as the conversational agent.
```
[the remark this answers is not in the document]
Comment (NoB0) [reply docs/source/configuration.rst]: Done.

File: docs/source/configuration.rst (hunk 1 above)
Comment (NoB0) [reply docs/source/configuration.rst]: Done.

File: docs/source/usage.rst (hunk 2 above)
Comment (NoB0) [reply docs/source/usage.rst]: Done.

File: docs/source/usage.rst (hunk 3 above)
Comment (NoB0) [reply docs/source/usage.rst]: DOne.

File: docs/source/usage.rst (hunk 4 above)
Comment (NoB0) [reply docs/source/usage.rst]: Done.

**Pull request merged**